pkg/cli/admin/nodeimage/common.go

```diff
@@ -0,0 +1,198 @@
+package nodeimage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/operator/resource/retry"
+	ocrelease "github.com/openshift/oc/pkg/cli/admin/release"
+	imagemanifest "github.com/openshift/oc/pkg/cli/image/manifest"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	kapierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	"k8s.io/kubectl/pkg/cmd/exec"
+)
+
+type CommonOptions struct {
+	genericiooptions.IOStreams
+	SecurityOptions imagemanifest.SecurityOptions
+
+	Config                   *rest.Config
+	remoteExecutor           exec.RemoteExecutor
+	ConfigClient             configclient.Interface
+	Client                   kubernetes.Interface
+	nodeJoinerImage          string
+	nodeJoinerNamespace      *corev1.Namespace
+	nodeJoinerServiceAccount *corev1.ServiceAccount
+	nodeJoinerRole           *rbacv1.ClusterRole
+	RESTClientGetter         genericclioptions.RESTClientGetter
+	nodeJoinerPod            *corev1.Pod
+	command                  string
+}
+
+func (o *CommonOptions) getNodeJoinerPullSpec(ctx context.Context) error {
```

Comment:
Please remove this common package entirely. This is not the preferred way and also I checked `getNodeJoinerPullSpec` (also `fetchClusterReleaseImage`) and monitor.go does not use this function. If the intend is `CommonOptions` will be a base for both commands, monitor command can use anything in create command as they are under the same `nodeimage` package. If there is strong necessity to have a base struct for both commands, as @andfasano suggested `BaseNodeImageCommand` can be used and it can be stored in nodeimage/create command rather than a new `common.go`

Reply:
I think that the main goal is to remove as much as possible the code duplication - which is a good thing - given the fact that the two command are almost very similar (as they use almost the same behavioral pattern). AFAICS though the package is still the `nodeimage` one, I'm more relaxed on having the common code in a separate file (or one of the command), as long as the naming is consistent (and the package remains the same ofc).

Reply:
I'm fine centralizing some of the components to prevent code duplication. I think, this can be achieved within `create` or `monitor` command instead of `common.go`.